### PR TITLE
WIP: Clippy returns!!

### PR DESCRIPTION
**Pull request by shaleh**

### CODE REVIEW

**Comment by shaleh** [comment]
@nickdrozd this is it.

File: rust_src/remacs-lib/docfile.rs [hunk 1]
```diff
@@ -27,12 +27,12 @@ const FUNCTION: c_int = 5;
 type AddGlobalFn = fn(c_int, *const c_char, c_int, *const c_char) -> *const ();
 
 #[no_mangle]
-pub extern "C" fn scan_rust_file(
+pub unsafe extern "C" fn scan_rust_file(
```
Comment (Wilfred) [inline rust_src/remacs-lib/docfile.rs]: I'm curious about this -- isn't it better to limit `unsafe` to the specific line?

Comment (shaleh) [reply rust_src/remacs-lib/docfile.rs]: Clippy says no.....

Comment (shaleh) [reply rust_src/remacs-lib/docfile.rs]: Which makes sense. If that `unsafe` line hits a snag the code is going to crash.

**Comment by brotzeit** [comment]
I guess we can close this PR ?

**Comment by shaleh** [comment]
Yeah, nick made a  new PR.

**Pull request closed without merging**